File: src/main/java/com/janboucek/crawler/neat/NodeGene.java
package com.janboucek.crawler.neat;

/**
 * Created by colander on 1/3/17.
 * NEAT node gene class.
 */
public class NodeGene {
    public final int innov;
    public final int type;
    public int activateFunction;

    public final static int TYPE_INPUT = 0;
    public final static int TYPE_OUTPUT = 1;
    public final static int TYPE_HIDDEN = 2;

    public final static int FUNCTION_SIGMOID = 0;
    public final static int FUNCTION_SIN = 1;
    public final static int FUNCTION_COS = 2;
    public final static int FUNCTION_LINEAR = 3;
    public final static int FUNCTION_ABS = 4;
    public final static int FUNCTION_GAUSS = 5;

    public final static int NO_FUNCTIONS = 6;

    public final static int FUNCTION_SHIFTED_SIGMOID = 7;

    NodeGene(int innov, int type, int activateFunction) {
        this.innov = innov;
        this.type = type;
        this.activateFunction = activateFunction;
    }
}

File: src/main/java/com/janboucek/crawler/neat/NodeGene.kt
package com.janboucek.crawler.neat

/**
 * Created by colander on 1/3/17.
 * NEAT node gene class.
 */
data class NodeGene constructor(val innov: Int, val type: Int, var activateFunction: Int) {
    companion object {
        const val TYPE_INPUT = 0
        const val TYPE_OUTPUT = 1
        const val TYPE_HIDDEN = 2

        const val FUNCTION_SIGMOID = 0
        const val FUNCTION_SIN = 1
        const val FUNCTION_COS = 2
        const val FUNCTION_LINEAR = 3
        const val FUNCTION_ABS = 4
        const val FUNCTION_GAUSS = 5
        const val NO_FUNCTIONS = 6
        const val FUNCTION_SHIFTED_SIGMOID = 7 // unused
    }
}